src/shapes/Object/Object.ts

```diff
@@ -1759,6 +1710,8 @@ export class FabricObject<
       'center'
     );
     const originalCanvas = this.canvas;
+    // static canvas and canvas have both an array of InteractiveObjects
+    // @ts-ignore this needs to be fixed somehow, or ignored globally
```

Comment:
good point

Reply:
I think static canvas should have the non interactive object as a type while canvas should have the interactive object
Wondering how complicated and dumb that might turn out